### PR TITLE
Add a saveMetric method to amber and gromacs process

### PR DESCRIPTION
**Pull request by xiki-tempula**
This PR condense the output to three pyarrow pandas dataframe file, which are the metric (potential/volume/...) and dHdl and u_nk files, which can be concatenated or modified in an engine agonist way.

### CODE REVIEW

File: .github/workflows/Sandpit_exs.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
 
       - name: Install dependency
         run: |
-          mamba install -c conda-forge -c openbiosim/label/main biosimspace python=3.8 ambertools gromacs "sire=2023.2.2" alchemlyb pytest "openff-interchange<0.3.0" "rdkit<2023"
+          mamba install -c conda-forge -c openbiosim/label/main biosimspace python=3.8 ambertools gromacs "sire=2023.2.2" alchemlyb pytest pyarrow openff-interchange "rdkit<2023"
```
Comment (msuruzhon) [inline .github/workflows/Sandpit_exs.yml]: Why do you need `pyarrow`?

Comment (xiki-tempula) [reply .github/workflows/Sandpit_exs.yml]: For pandas serialisation. Do you not like pyarrow?

Comment (msuruzhon) [reply .github/workflows/Sandpit_exs.yml]: I don't really have an opinion, was just wondering why you don't use `pickle` or a `csv`?

Comment (xiki-tempula) [reply .github/workflows/Sandpit_exs.yml]: csv won't preserve the index. Pickle is fine until something changes and you are screwed. parquet via pyarrow is the pandas recommended serialisation method, which shall preserve the index as it is. It is fast and has built in compression.

File: tests/Sandpit/Exscientia/Process/test_gromacs.py [hunk 2]
```diff
@@ -274,3 +282,19 @@ def test_get(self, setup, func, time_series, value, unit):
...
         else:
             np.testing.assert_almost_equal(energy / unit, value, decimal=3)
+
+    def test_metric_parquet_exist(self, setup):
+        assert Path(f"{setup.workDir()}/metric.parquet").exists()
+
+    def test_dhdl_parquet_exist(self, setup):
+        assert Path(f"{setup.workDir()}/dHdl.parquet").exists()
+
+    def test_u_nk_parquet_exist(self, setup):
```
Comment (msuruzhon) [inline tests/Sandpit/Exscientia/Process/test_gromacs.py]: Also check the dimensionalities of `dhdl` and `u_nk`?

File: python/BioSimSpace/Sandpit/Exscientia/Process/_amber.py [hunk 3]
```diff
@@ -2481,3 +2484,36 @@ def _get_stdout_record(self, key, time_series=False, unit=None, dof=0):
 
             except KeyError:
                 return None
+
+    def saveMetric(
+        self, filename="metric.parquet", u_nk="u_nk.parquet", dHdl="dHdl.parquet"
+    ):
```
Comment (msuruzhon) [inline python/BioSimSpace/Sandpit/Exscientia/Process/_amber.py:2490]: Some docstrings would be good

**Review by msuruzhon — APPROVED** [review]
LGTM, only have small comments

**Pull request merged**